### PR TITLE
Migrate wazuh cypress

### PR DESCRIPTION
**Pull request by matiasmoreno876**
Hi team in this PR I migrate all the development done with cypress to our wazuh-app project.
New workflows for image building and cypress test execution have also been added.

closes [#4031](https://github.com/wazuh/wazuh-kibana-app/issues/4031)

### CODE REVIEW

File: .github/workflows/cypress-basic.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        echo "Information about the workflow (basic):
+        wazuh-version: ${{ github.event.inputs.wazuh-version }}
+        elastic-version: ${{ github.event.inputs.elastic-version }}
+        ubuntu-cypress-version: ${{ github.event.inputs.ubuntu-cypress-version }}
+        url-package: ${{ github.event.inputs.url-package }}"
+        
+     - name: Step 02 - Download Project wazuh-app-environments 
+       uses: actions/checkout@v2
+       with:
+        repository: matiasmoreno876/wazuh-app-environments
```
Comment (Desvelao) [inline .github/workflows/cypress-basic.yml:47]: **thought**: we should have the environments in this repository.

File: .github/workflows/cypress-basic.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        echo "TEST_PASSING=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-3)}')" >> $GITHUB_ENV
+        echo "TEST_TOTAL=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-4)}')" >> $GITHUB_ENV
+        echo "TEST_TIME=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-5)}')" >> $GITHUB_ENV
+     - name: Step 11 - Post to a Slack channel
+       id: slack
+       uses: slackapi/slack-github-action@v1.19.0
+       with:
+          # Slack channel id, channel name, or user id to post message.
+          # See also: https://api.slack.com/methods/chat.postMessage#channels
+        channel-id: 'daily-app-team'
```
Comment (Desvelao) [inline .github/workflows/cypress-basic.yml:110]: **thought**: this should be a secret.

File: .github/workflows/cypress-odfe.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        echo "TEST_PASSING=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-3)}')" >> $GITHUB_ENV
+        echo "TEST_TOTAL=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-4)}')" >> $GITHUB_ENV
+        echo "TEST_TIME=$(grep -E "failed \(|passed" cypress.log | awk ' { print $(NF-5)}')" >> $GITHUB_ENV
+     - name: Step 11 - Post to a Slack channel
+       id: slack
+       uses: slackapi/slack-github-action@v1.19.0
+       with:
+          # Slack channel id, channel name, or user id to post message.
+          # See also: https://api.slack.com/methods/chat.postMessage#channels
+        channel-id: 'daily-app-team'
```
Comment (Desvelao) [inline .github/workflows/cypress-odfe.yml:116]: **thought**: this should be a secret

File: cypress/cypress/integration/features/management/decoders/edit-decoders-without-saving.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Feature: Should Displayes a warning modal
```
Comment (Desvelao) [inline cypress/cypress/integration/features/management/decoders/edit-decoders-without-saving.feature:1]: **nitpick**: Typo: `Displayes`

**Comment by Desvelao** [comment]
**review**:
Code review: I see some things we should remove:
- .DS_Store file/s
- Test reporting (videos, images, etc...)

For another hand, I see we are using Docker images of a team developer instead of the organization. Do we want this for now?

File: .github/workflows/cypress-basic.yml (hunk 1 above)
Comment (matiasmoreno876) [reply .github/workflows/cypress-basic.yml:47]: Yes, for the time being we will continue to use them in this way until the new environments are integrated into the project.

File: .github/workflows/cypress-basic.yml (hunk 2 above)
Comment (matiasmoreno876) [reply .github/workflows/cypress-basic.yml]: Done!

File: .github/workflows/cypress-odfe.yml (hunk 3 above)
Comment (matiasmoreno876) [reply .github/workflows/cypress-odfe.yml]: Done!

**Comment by matiasmoreno876** [comment]
The .DS_Store and Test reporting files have already been deleted, thank you very much for the observation @Desvelao 
Regarding docker images, they were previously consumed from here [docker hub](https://hub.docker.com/search?q=matiasmoreno876). 
this project was modified and the images are consumed from here [quay.io repository](https://quay.io/repository/wazuh/wazuh-manager-image?tab=tags)

**Review by Desvelao — CHANGES_REQUESTED** [review]
**nitpick**: There is a `.DS_Store` file, we should remove it.

**Comment by matiasmoreno876** [comment]
@Desvelao, `.DS_Store` file was delete

**Comment by mauceballos** [comment]
## xpack testing

### summary

<details>

Screenshot from 2022-09-01 16-31-59

Screenshot from 2022-09-01 14-35-39

</details>

### Report

<details>

screencapture-file-home-mauri-repo-kibana-app-wazuh-kibana-app-test-cypress-cypress-report-output-html-2022-09-01-16_32_26

</details>

**Comment by mauceballos** [comment]
## wzd testing

### summary

<details>

Screenshot from 2022-09-02 09-47-14


</details>

### Report

<details>

screencapture-file-home-mauri-repo-kibana-app-wazuh-kibana-app-test-cypress-cypress-report-output-html-2022-09-02-09_44_03


</details>

**Review by asteriscos — APPROVED** [review]
LGTM :heavy_check_mark: 

Screenshot from 2022-09-02 15-56-31

**Pull request merged**